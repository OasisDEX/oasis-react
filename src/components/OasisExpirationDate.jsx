import React, { PureComponent } from "react";
import { PropTypes } from "prop-types";
// import ImmutablePropTypes from 'react-immutable-proptypes';

import styles from "./OasisExpirationDate.scss";
import CSSModules from "react-css-modules";
import * as moment from "moment";
import OasisLoadingIndicator from "./OasisLoadingIndicator";

const propTypes = PropTypes && {
  timestamp: PropTypes.number.isRequired
};
const defaultProps = {};

const formatTime = ts =>
  ts ? (
    moment.unix(ts).format("DD-M-YYYY")
  ) : (
    <div style={{ marginLeft: 20 }}>
      <OasisLoadingIndicator />
    </div>
  );

export class OasisExpirationDate extends PureComponent {
  render() {
    const { timestamp } = this.props;
    return (
      <div styleName="ClosingTime">
        <div> CLOSING TIME </div>
        <div styleName="Date">{formatTime(timestamp)}</div>
      </div>
    );
  }
}

OasisExpirationDate.displayName = "OasisExpirationDate";
OasisExpirationDate.propTypes = propTypes;
OasisExpirationDate.defaultProps = defaultProps;

export default CSSModules(OasisExpirationDate, styles);
