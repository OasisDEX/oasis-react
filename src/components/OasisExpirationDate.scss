@import "../styles/modules/labels";

.ClosingTime {
  @include text(12px);

  color: #4F5362;

  letter-spacing: 1.3px;
  text-transform: uppercase;

  & .Date {
    display: flex;
    margin-top: 5px;
    align-items: center;
    @include text(13px);
    color: rgba(79, 83, 98, 0.8);

    &:before {
      content: "";
      width: 13px;
      height: 13px;
      margin-right: 5px;
      background: url("../assets/clock.svg");
    }
  }
}