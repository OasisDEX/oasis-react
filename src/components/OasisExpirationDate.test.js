/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import { OasisExpirationDate } from './OasisExpirationDate';

describe('(Component) OasisExpirationDate', () => {
  it('should render', () => {
    const props = {
      timestamp: 1536674743
    };
    const wrapper = shallow(
      <OasisExpirationDate {...props}/>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
