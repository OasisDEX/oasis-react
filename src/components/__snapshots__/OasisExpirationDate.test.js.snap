// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisExpirationDate should render 1`] = `
<div
  styleName="ClosingTime"
>
  <div>
     CLOSING TIME 
  </div>
  <div
    styleName="Date"
  >
    11-9-2018
  </div>
</div>
`;
